### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by JoePlayz** [genboy/Festival#108]
I turned on PvP flag and when I tried hitting a player that had a PvP flag on, it showed Internal server error.

here's my console log

`29.12 13:37:46 [Server] Server thread/CRITICAL Error: "Call to undefined method genboy\Festival\lang\Language::tranlate()" (EXCEPTION) in "Festival.phar/src/genboy/Festival/Main" at line 1783
29.12 13:37:46 [Server] Server thread/DEBUG #0 Festival.phar/src/genboy/Festival/Main(1806): genboy\Festival\Main->canPVP(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #1 Festival.phar/src/genboy/Festival/Main(1470): genboy\Festival\Main->canDamage(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #2 src/pocketmine/plugin/MethodEventExecutor(38): genboy\Festival\Main->onHurt(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #3 src/pocketmine/plugin/RegisteredListener(98): pocketmine\plugin\MethodEventExecutor->execute(genboy\Festival\Main object, pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #4 src/pocketmine/event/Event(96): pocketmine\plugin\RegisteredListener->callEvent(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #5 src/pocketmine/entity/Entity(929): pocketmine\event\Event->call()
29.12 13:37:46 [Server] Server thread/DEBUG #6 src/pocketmine/entity/Living(559): pocketmine\entity\Entity->attack(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #7 src/pocketmine/Player(3651): pocketmine\entity\Living->attack(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #8 src/pocketmine/Player(2504): pocketmine\Player->attack(pocketmine\event\entity\EntityDamageByEntityEvent object)
29.12 13:37:46 [Server] Server thread/DEBUG #9 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(139): pocketmine\Player->handleInventoryTransaction(pocketmine\network\mcpe\protocol\InventoryTransactionPacket object)
29.12 13:37:46 [Server] Server thread/DEBUG #10 src/pocketmine/network/mcpe/protocol/InventoryTransactionPacket(155): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleInventoryTransaction(pocketmine\network\mcpe\protocol\InventoryTransactionPacket object)
29.12 13:37:46 [Server] Server thread/DEBUG #11 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(99): pocketmine\network\mcpe\protocol\InventoryTransactionPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
29.12 13:37:46 [Server] Server thread/DEBUG #12 src/pocketmine/network/mcpe/protocol/BatchPacket(114): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\InventoryTransactionPacket object)
29.12 13:37:46 [Server] Server thread/DEBUG #13 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(99): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\network\mcpe\PlayerNetworkSessionAdapter object)
29.12 13:37:46 [Server] Server thread/DEBUG #14 src/pocketmine/Player(3044): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
29.12 13:37:46 [Server] Server thread/DEBUG #15 src/pocketmine/network/mcpe/RakLibInterface(162): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
29.12 13:37:46 [Server] Server thread/DEBUG #16 vendor/pocketmine/raklib/src/server/ServerHandler(98): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 183.90.36.103 9701, raklib\protocol\EncapsulatedPacket object, integer 0)
29.12 13:37:46 [Server] Server thread/DEBUG #17 src/pocketmine/network/mcpe/RakLibInterface(102): raklib\server\ServerHandler->handlePacket()
29.12 13:37:46 [Server] Server thread/DEBUG #18 src/pocketmine/network/Network(94): pocketmine\network\mcpe\RakLibInterface->process()
29.12 13:37:46 [Server] Server thread/DEBUG #19 src/pocketmine/network/mcpe/RakLibInterface(92): pocketmine\network\Network->processInterface(pocketmine\network\mcpe\RakLibInterface object)
29.12 13:37:46 [Server] Server thread/DEBUG #20 vendor/pocketmine/snooze/src/SleeperHandler(120): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
29.12 13:37:46 [Server] Server thread/DEBUG #21 vendor/pocketmine/snooze/src/SleeperHandler(82): pocketmine\snooze\SleeperHandler->processNotifications()
29.12 13:37:46 [Server] Server thread/DEBUG #22 src/pocketmine/Server(2278): pocketmine\snooze\SleeperHandler->sleepUntil(double 1546090666.7955)
29.12 13:37:46 [Server] Server thread/DEBUG #23 src/pocketmine/Server(2139): pocketmine\Server->tickProcessor()
29.12 13:37:46 [Server] Server thread/DEBUG #24 src/pocketmine/Server(1715): pocketmine\Server->start()
29.12 13:37:46 [Server] Server thread/DEBUG #25 src/pocketmine/PocketMine(249): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string /, string /plugins/)
29.12 13:37:46 [Server] Server thread/DEBUG #26 /jar/PocketMine-MP_24_dec.phar(1): require(string phar:///jar/PocketMine-MP_24_dec.phar/src/pocketmine/PocketMine.php)`

**Comment by genboy** [comment]
Thank you for reporting this, i'll check to reproduce and try to fix it.

**Comment by genboy** [comment]
@JoePlayz fixed the error and some other typo's, dev phar version is available for testing at [poggit.pmmp.io](https://poggit.pmmp.io/ci/recent)

**Comment by JoePlayz** [comment]
It works! Thanks. However, is there a way where I could only enable fly in areas I set? Because if I disable fly in the config of a certain world, I can’t fly in the entire world even those that are not area protected and if I enable fly in the config, players are able to fly everywhere even the areas that are not area protected. Also, I found another typo, in the /fe here and /fe list, you mistyped attached as “attachted”. Thanks for your hard work
